### DISCUSSION TITLE
Why zepeto TypeScript script can't get bones

### DISCUSSION THREAD

**Discussion by HotObsidian** [naverz/zepeto-studio-global#74] [Q&A]
Recently, when I rewrote the c# script into a Typescript script, I found that the bones in the SkinnedMeshRenderer component could not be obtained, and all the members in the bones array were undefined. What is the problem?

**Comment by HotObsidian** [comment]
let Bones: Transform[] = this.Target.GetComponent<SkinnedMeshRenderer>().bones;
 Debug.Log("Bones" + Bones[1]);

I wrote it like this and the result is: Bones undefined

**Reply by naverz-mkao** [reply]
Hi HotObsidian,

You will NOT be able to do GetComponent().bones.

If you look at the TypeScript definitions, there is no "GetComponent()" that takes 0 arguments.

Here's a sample code of grabbing the bones using "GetComponentInChildren".

```import { Debug, SkinnedMeshRenderer } from 'UnityEngine';
import { LocalPlayer, SpawnInfo, ZepetoCharacter, ZepetoCharacterCreator, ZepetoPlayer, ZepetoPlayers } from 'ZEPETO.Character.Controller'
import { ZepetoScriptBehaviour } from 'ZEPETO.Script'

export default class PlayerLoader extends ZepetoScriptBehaviour {

    Start() {    
 
        ZepetoPlayers.instance.CreatePlayerWithZepetoId("", "[ZEPETO_ID]", new SpawnInfo(), 
        true);

        ZepetoPlayers.instance.OnAddedLocalPlayer.AddListener(() => {
                let _player : LocalPlayer = ZepetoPlayers.instance.LocalPlayer;
                Debug.LogError(_player.zepetoPlayer.character.GetComponentInChildren<SkinnedMeshRenderer>().bones[0].name);
            });
        }

}

Hope this helps!

**Reply by HotObsidian** [reply]
In fact, I made a mistake when I asked the question. I wrote GetComponentInChildren< SkinnedMeshRenderer >().bones, and then I copied your code. When the character is loaded normally, the debug result is still undefined. then I added the code you gave me to the official zepeto project, and the result is still undefined too

**Reply by HotObsidian** [reply]
I found a way. Although bones are arrays, we cannot use [i] to output array members, we can only rely on getItem(i)

**Comment by naverz-mkao** [comment]
Hi HotObsidian,

Ah yes sorry about that. I just realized the issue is that you will need to be on 1.6.0 to be able to output array members using the [] notation.

<img width="487" alt="image" src="https://user-images.githubusercontent.com/107931355/190855125-7a68ffd5-ac52-4420-b63c-fc876e84bf72.png">

Before 1.6.0, you are correct, you have to use get_Item(0) to output array members. You can NOT use [i] to output array members.
We fixed that issue with the 1.6.0 update. With the 1.6.0 update, you will be able to use [i] to output array members.

Zepeto World 1.6.0 update will also give you many other enhancements such as improved camera collision and console debugging on mobile devices! You can read about the 1.6.0 update here: https://studio.zepeto.me/guides/world-160.

As always, please make sure to make a backup of your project before upgrading the ZEPETO.World package.

Sorry for the confusion.

**Reply by HotObsidian** [reply]
ok，thx~
